crates/subspace-farmer/src/plot.rs

```diff
@@ -290,8 +304,43 @@ impl WeakPlot {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct TypedDB<K, V> {
+    inner: DB,
+    _kv: PhantomData<(K, V)>,
+}
+
+impl<K, V> From<DB> for TypedDB<K, V> {
+    fn from(inner: DB) -> Self {
+        let _kv = PhantomData;
+        Self { inner, _kv }
+    }
+}
+
+impl TypedDB<u64, u64> {
+    pub fn get(&self, key: u64) -> io::Result<Option<u64>> {
+        self.inner
+            .get(key.to_be_bytes())
+            .map_err(io::Error::other)
+            .and_then(|opt_val| {
+                opt_val
+                    .map(|val| TryInto::<[u8; 8]>::try_into(val).map(u64::from_be_bytes))
+                    .transpose()
+                    .map_err(|_| io::Error::other("Offsets in rocksdb supposed to be 8 bytes long"))
+            })
+    }
+
+    pub fn put(&self, key: u64, value: u64) -> io::Result<()> {
+        self.inner
+            .put(key.to_be_bytes(), value.to_be_bytes())
+            .map_err(io::Error::other)
+    }
+}
```

Comment:
I don't really see why this abstraction is necessary

Reply:
Just to remove some duplicate code from getting and putting keys, as it is used twice.

Reply:
This is not the case right now, but it just makes code cleaner and more self-describing